packages/core/src/accessanalyzer/vue/iamPolicyChecks.ts

```diff
@@ -0,0 +1,146 @@
+        public readonly onChangeCloudformationParameterFilePath = new vscode.EventEmitter<string>()
+    ) {
+        super(IamPolicyChecksWebview.sourcePath)
+        this._setActiveTextEditorListener()
+        this._setActiveConfigurationListener()
+    }
+
+    public init(): typeof this.data {
+        return this.data
+    }
+
+    public async getReferenceDocument(path: string): Promise<string> {
+        return _getReferenceDocument(path)
+    }
+
+    public _setActiveTextEditorListener() {
+        // Send the current active text editor to Webview to show what is being targeted by the user
+        vscode.window.onDidChangeActiveTextEditor((message: any) => {
+            const editedFile = vscode.window.activeTextEditor?.document
+            this.onChangeInputPath.fire(editedFile!.uri.path)
+        })
+    }
+    public _setActiveConfigurationListener() {
+        vscode.workspace.onDidChangeConfiguration((config: vscode.ConfigurationChangeEvent) => {
+            // If settings change, we want to update the Webview to reflect the change in inputs
+            if (config.affectsConfiguration('aws.accessAnalyzer.policyChecks.referencePolicyFilePath')) {
+                this.onChangeReferenceFilePath.fire(
+                    vscode.workspace.getConfiguration().get('aws.accessAnalyzer.policyChecks.referencePolicyFilePath')!
+                )
+            } else if (config.affectsConfiguration('aws.accessAnalyzer.policyChecks.cloudFormationParameterFilePath')) {
+                this.onChangeCloudformationParameterFilePath.fire(
+                    vscode.workspace
+                        .getConfiguration()
+                        .get('aws.accessAnalyzer.policyChecks.cloudFormationParameterFilePath')!
+                )
+            }
+        })
+    }
+
+    // TODO: Add functionality for the three checks
+    public async validatePolicy(documentType: string, policyType: string) {
+        this.client /* Do something */
+    }
+    public async checkNoNewAccess() {}
+    public async checkAccessNotGranted() {}
+}
+
+const Panel = VueWebview.compilePanel(IamPolicyChecksWebview)
+
+export async function renderIamPolicyChecks(context: ExtContext): Promise<void> {
+    const logger: Logger = getLogger()
+    try {
+        const client = new AccessAnalyzer({ region: context.regionProvider.defaultRegionId })
+        //Read from settings to auto-fill some inputs
+        const referencePolicyFilePath: string = vscode.workspace
+            .getConfiguration()
+            .get('aws.accessAnalyzer.policyChecks.referencePolicyFilePath')!
+        const cfnParameterPath: string = vscode.workspace
+            .getConfiguration()
+            .get('aws.accessAnalyzer.policyChecks.cloudFormationParameterFilePath')!
+
+        //Check if Cfn and Tf tools are installed
+        var cfnToolInstalled = true
+        var tfToolInstalled = true
+        exec('tf-policy-validator', err => {
```

Comment:
Q: do we plan to surface other errors from the CLI tool to VS code customers?

Reply:
Yes, that will be included when I add the CLI implementations.